mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -0,0 +1,793 @@
+package mongodbatlas
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/go-test/deep"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mwielbut/pointy"
+	"log"
+	"strings"
+
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasSearchIndex() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceMongoDBAtlasSearchIndexCreate,
+		Read:   resourceMongoDBAtlasSearchIndexRead,
+		Update: resourceMongoDBAtlasSearchIndexUpdate,
+		Delete: resourceMongoDBAtlasSearchIndexDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMongoDBAtlasSearchIndexImportState,
+		},
+		Schema: returnSearchIndexSchema(),
+	}
+}
+
+func returnSearchIndexSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"project_id": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"cluster_name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"index_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+			Required: false,
+		},
+		"analyzer": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"analyzers": customAnalyzersSchema(),
+		"collection_name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"database": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"search_analyzer": {
+			Type:     schema.TypeString,
+			Required: false,
+		},
+		"mappings_dynamic": {
+			Type:     schema.TypeBool,
+			Required: false,
+		},
+		"mappings_fields": {
```

Comment:
this is another parameter that can include a lot of data in, fyi

Reply:
this field `mappings_fields` needs to be in a JSON format in order to get correctly this property (because the `mapping_field` structure contains itself and cannot be describe as an schema in provider definition (it will take `n` sublevels for mapping_fields)